Add Final classes into UPGRADE-4.0.md (#739)
sonata-project/SonataClassificationBundle: UPGRADE-4.0.md

 * `getAllRootCategories`
 * `getRootCategoriesSplitByContexts`

## Final classes

> Some classes  are now `final` and should not be overridden:

Reviewer: ```suggestion
Some classes are declared as `final` and MUST NOT be overridden:
```
